src/main/java/seedu/address/logic/commands/testresult/EditTestResultCommand.java

```diff
@@ -0,0 +1,174 @@
+package seedu.address.logic.commands.testresult;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.CommandType;
+import seedu.address.logic.commands.EditCommand;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.patient.Nric;
+import seedu.address.model.testresult.MedicalTest;
+import seedu.address.model.testresult.Result;
+import seedu.address.model.testresult.TestDate;
+import seedu.address.model.testresult.TestResult;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.*;
+
+/**
+ * Edits the details of an existing test result information in MedBook.
+ */
+public class EditTestResultCommand extends Command {
+    public static final String COMMAND_WORD = "edit";
+    public static final CommandType COMMAND_TYPE = CommandType.TEST;
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edits the details of the medical information identified "
```

Comment:
Should it be test result?
```suggestion
            + ": Edits the details of the test result identified "
```